feat(conversations): added conversations package (#905)
Vonage/vonage-node-sdk: packages/conversations/README.md

# Vonage Conversations SDK for Node.js

![GitHub Workflow Status (branch)](https://img.shields.io/github/actions/workflow/status/Vonage/vonage-node-sdk/ci.yml?branch=3.x) [![Codecov](https://img.shields.io/codecov/c/github/vonage/vonage-node-sdk?label=Codecov&logo=codecov&style=flat-square)](https://codecov.io/gh/Vonage/vonage-server-sdk) ![Latest Release](https://img.shields.io/github/v/release/vonage/vonage-node-sdk?logo=npm&style=flat-square) [![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-v2.0%20adopted-ff69b4.svg?style=flat-square)](../../CODE_OF_CONDUCT.md) [![License](https://img.shields.io/npm/l/@vonage/server-sdk?label=License&style=flat-square)](../../LICENSE.TXT)

<img src="https://developer.nexmo.com/images/logos/vbc-logo.svg" height="48px" alt="Vonage" />

This is the Vonage Conversations SDK for Node.js for use with [Vonage APIs](https://www.vonage.com/). To use it you will need a Vonage account. Sign up [for free][signup] at vonage.com.

> For full API documentation refer to [developer.nexmo.com](https://developer.nexmo.com/).

Reviewer: ```suggestion
For full API documentation refer to [Vonage Conversation API Overview](https://developer.vonage.com/en/conversation/overview).
```

If you are updating from V2 to V3, please check the migration guide found [here](https://github.com/Vonage/vonage-node-sdk/blob/3.x/packages/conversations/v2_TO_v3_MIGRATION_GUIDE.md)

* [Installation](#installation)
* [Usage](#usage)
* [Promises](#promises)

## Installation

We recommend using this SDK as part of the overall [`@vonage/server-sdk` package](https://github.com/vonage/vonage-node-sdk). Please see the main package for installation.

You can also use this SDK standalone if you only need access to just the Conversations API.

### With NPM

```bash
npm install @vonage/conversations
```

### With Yarn

```bash
yarn add @vonage/conversations
```

## Usage

### As part of the Vonage Server SDK

If you are using this SDK as part of the Vonage Server SDK, you can access it as the `conversations` property off of the client that you instantiate.

```js
const {Vonage} = require('@vonage/server-sdk');

> const { Auth, AlgorithmTypes } = require('@vonage/auth');

Reviewer: ```suggestion
const { Auth } = require('@vonage/auth');
```
Don't see AlgorithmTypes being used in the snippet.

> const { SMS } = require('@vonage/conversations');

Reviewer: ```suggestion
```
Is this needed and why does it say SMS?

const vonage = new Vonage(new Auth({
  apiKey: API_KEY,
  apiSecret: API_SECRET,
  applicationId: APP_ID,
  privateKey: PRIVATE_KEY_PATH,
}), options);

vonage.conversations.getConversation(CONVERSATION_ID)
```

### Standalone

The SDK can be used standalone from the main [Vonage Server SDK for Node.js](https://github.com/vonage/vonage-node-sdk)
if you only need to use the Conversations API. All you need to do
is `require('@vonage/conversations')`, and use the returned object to create your own
client.

```js
const {Auth} = require('@vonage/auth');
const {Conversations} = require('@vonage/conversations');

const conversationsClient = new Conversations(new Auth({
  apiKey: API_KEY,
  apiSecret: API_SECRET,
  applicationId: APP_ID,
  privateKey: PRIVATE_KEY_PATH,

> }), options);

Reviewer: ```suggestion
}), options);

conversationsClient.getConversation(CONVERSATION_ID)
```

> ```

Reviewer: ```suggestion

conversationsClient.getConversation(CONVERSATION_ID)
```
```
Did we want to add this to be consistent with the other snippet?
